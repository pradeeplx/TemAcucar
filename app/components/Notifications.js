import React, { Component } from 'react'
import { View, ActivityIndicator } from 'react-native'
import LoadMore from "../components/LoadMore"
import Notification from "../components/Notification"

export default Notifications = ({ notifications, listing, canList, onList, onView }) => (
  <View >
    { notifications.map((notification) => (
      <Notification
        key={notification.id}
        notification={notification}
        onView={onView}
      />
    )) }
    { listing && <ActivityIndicator style={{ marginTop: 10 }} /> }
    { canList && !listing &&
      <LoadMore onPress={onList} />
    }
  </View>
)
