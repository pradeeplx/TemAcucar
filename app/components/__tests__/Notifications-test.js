import 'react-native'
import React from 'react'
import Notifications from '../Notifications'
import renderer from 'react-test-renderer'
import { identity } from 'lodash'

const props = {
  notifications: [
    { id: 1 },
    { id: 2 },
  ]
}

it('renders correctly', () => {
  const tree = renderer.create(
    <Notifications {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ActivityIndicator', () => {
  const tree = renderer.create(
    <Notifications {...props} listing />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders load more button', () => {
  const tree = renderer.create(
    <Notifications {...props} canList />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
