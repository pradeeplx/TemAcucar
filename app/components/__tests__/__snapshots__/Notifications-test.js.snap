exports[`test renders ActivityIndicator 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FAF6F1",
        "borderBottomWidth": 0.5,
        "borderColor": "#CECED2",
        "opacity": 1,
        "padding": 10,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              undefined,
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail" />
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              Array [
                Object {
                  "fontSize": 13.5,
                },
                Object {
                  "color": "#CECED2",
                  "marginTop": 4,
                },
              ],
            ]
          }>
          H치 segundos
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FAF6F1",
        "borderBottomWidth": 0.5,
        "borderColor": "#CECED2",
        "opacity": 1,
        "padding": 10,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              undefined,
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail" />
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              Array [
                Object {
                  "fontSize": 13.5,
                },
                Object {
                  "color": "#CECED2",
                  "marginTop": 4,
                },
              ],
            ]
          }>
          H치 segundos
        </Text>
      </View>
    </View>
  </View>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
    style={
      Object {
        "marginTop": 10,
      }
    } />
</View>
`;

exports[`test renders correctly 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FAF6F1",
        "borderBottomWidth": 0.5,
        "borderColor": "#CECED2",
        "opacity": 1,
        "padding": 10,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              undefined,
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail" />
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              Array [
                Object {
                  "fontSize": 13.5,
                },
                Object {
                  "color": "#CECED2",
                  "marginTop": 4,
                },
              ],
            ]
          }>
          H치 segundos
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FAF6F1",
        "borderBottomWidth": 0.5,
        "borderColor": "#CECED2",
        "opacity": 1,
        "padding": 10,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              undefined,
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail" />
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              Array [
                Object {
                  "fontSize": 13.5,
                },
                Object {
                  "color": "#CECED2",
                  "marginTop": 4,
                },
              ],
            ]
          }>
          H치 segundos
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`test renders load more button 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FAF6F1",
        "borderBottomWidth": 0.5,
        "borderColor": "#CECED2",
        "opacity": 1,
        "padding": 10,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              undefined,
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail" />
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              Array [
                Object {
                  "fontSize": 13.5,
                },
                Object {
                  "color": "#CECED2",
                  "marginTop": 4,
                },
              ],
            ]
          }>
          H치 segundos
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FAF6F1",
        "borderBottomWidth": 0.5,
        "borderColor": "#CECED2",
        "opacity": 1,
        "padding": 10,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              undefined,
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail" />
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              Array [
                Object {
                  "fontSize": 13.5,
                },
                Object {
                  "color": "#CECED2",
                  "marginTop": 4,
                },
              ],
            ]
          }>
          H치 segundos
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#654C50",
            },
            Array [
              Object {
                "marginBottom": 20,
                "marginTop": 10,
                "textAlign": "center",
              },
              undefined,
            ],
          ],
        ]
      }>
      Carregar mais
    </Text>
  </View>
</View>
`;
